cpio/cpio.c

```diff
@@ -1080,7 +1080,11 @@ extract_data(struct archive *ar, struct archive *aw)
 			exit(1);
 		}
 		r = (int)archive_write_data_block(aw, block, size, offset);
```

Comment:
I'm a little nervous about this.  Should we change the return value of `archive_write_data_block`?   Looking at the implementation, it doesn't look like the proposed 4.0 version of `archive_write_data_block` would ever return a value other than the requested size, so it might make more sense to have `archive_write_data_block` continue to return `ARCHIVE_OK` on success.  That would be less disruptive to anyone who is actually using this interface.

That would instead suggest that we:
* Change the documentation to match the implementation (instead of the other way around)
* Maybe change the return type to a simple `int`

Reply:
I'll remove the proposed 4.0 change to `archive_write_data_block`'s return value and update the documentation in consequence then.

Reply:
Thank you.  I think that's probably the right thing to do here.